xla/service/gpu/nccl_api.cc

```diff
@@ -522,8 +537,135 @@ DefaultNcclApi::DeregisterBuffer(NcclCommHandle comm,
   VLOG(3) << absl::StreamFormat(
       "Deregister buffer for NCCL communicator; handle=%p; comm=%p", handle,
       comm);
+#if (NCCL_VERSION_CODE >= 21901)    
   return XLA_NCCL_STATUS(
       ncclCommDeregister(Cast(comm), reinterpret_cast<void*>(handle)));
+#endif // NCCL_VERSION_CODE >= 21901
+}
+
+#else // XLA_ENABLE_XCCL
```

Comment:
I'm trying to delete `XLA_ENABLE_XCCL` ifdef from XLA code base, and new coding style is "no new ifdefs", the plan to is keep "real" implementation completely separate from a stub, albeit it adds some extra complexity to build files.

We certainly don't want large blocks of code undef ifdef as it breaks all our static analysis tooling. It's ok to have small regions under ifdef to define aliases, but not in the middle of the file.

Reply:
But that means we will need a if_gpu_is_configured macro that supports an else branch.

Reply:
Right now, if_gpu_is_configured is if_cuda_is_configured(x) + if_rocm_is_configured(x). For if_cuda_or_rocm, we simply specify a select that is the union of if_cuda and if_rocm macro. For if_gpu_is_configured, that is not easily possible, as if_cuda_is_configured and if_rocm_is_configured are in template .bzl files and do something differently based on build variables: "if %{rocm_is_configured}" "if %{cuda_is_configured}". I think we would need to adapt cuda_configure.bzl and rocm_configure.bzl to specify both variables, so that we can make use of them.

Reply:
And thinking more about it, I don't know how this can work. Maybe someone can figure this out, but it will be very tricky.

Reply:
I am trying https://github.com/openxla/xla/pull/9266 now, which loads enable_cuda from cuda_configure.bzl and then sets a if_gpu_is_configured macro to false only if also enable_cuda returns false. So we can have the if_gpu_is_configured macro in the rocm:build_defs.bzl file, and re-export if via the stream_executor build_defs.bzl file.

Reply:
The PR has been merged now, can you please try to switch to if_gpu_is_configured macro and keep the stub file?

Reply:
Thanks for the help! Done.